### ISSUE TITLE
Separate Tor circuits for checking subscriptions

### ISSUE THREAD

**Issue by user234683** [user234683/youtube-local#16]
Subscriptions are unique per user. They should be done in a separate circuit so they cannot be associated with what videos are being watched. Need to figure out how to do this, preferably without bundling Tor ourselves.

Research:
- Are any special APIs provided by the Tor Socks5 proxy? Does "Tor RC" have anything to do with it or is that strictly the configuration file?
- If not, what libraries/APIs are available and what is the recommended way to integrate with them?

**Comment by zrose584** [comment]
I would expect tor to do this (to a certain degree) by default. Ofc one should use seperate tcp connections for seperate subscription feeds.

If you want more control over the tor-stream-id -> tor-circuit-id mapping, you can add a hook like this:
```python
import stem.control

def attach_stream(self, stream):
    if stream.status == 'NEW':
        dst_cid = more_logic()
        print(f"mapping stream to circuit {dst_cid}")
        controller.attach_stream(stream.id, dst_cid)

controller = stem.control.Controller.from_port(port=my_control_port)
controller.add_event_listener(attach_stream, stem.control.EventType.STREAM)
```

https://stem.torproject.org/api/control.html#stem.control.Controller.attach_stream

Instead of tor you could also use proxies.. There is a library which fits this usecase:
https://github.com/constverum/ProxyBroker/blob/master/examples/find_and_use.py

**Comment by user234683** [comment]
Awesome, this looks exactly like what we need. I'll get around to integrating this

**Comment by user234684** [comment]
>     * Are any special APIs provided by the Tor Socks5 proxy?
Yes, the Tor Socks5 proxy separates circuits if they are authenticated with different accounts. The channel ID could be included there: `socks5h://channel_id:channel_id@127.0.0.1:tor_port`
